pkg/hintrunner/zero/zerohint_uint512.go

```diff
@@ -0,0 +1,102 @@
+package zero
+
+import (
+	"math/big"
+
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+)
+
+const (
+	P_LOW  = "201385395114098847380338600778089168199"
+	P_HIGH = "64323764613183177041862057485226039389"
+)
+
+func newInvModPUint512Hint(x, xInverseModP hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "InvModPUint512",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> def pack_512(u, num_bits_shift: int) -> int:
+			//>     limbs = (u.d0, u.d1, u.d2, u.d3)
+			//>     return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+			//>
+			//> x = pack_512(ids.x, num_bits_shift = 128)
+			//> p = ids.p.low + (ids.p.high << 128)
+			//> x_inverse_mod_p = pow(x,-1, p)
+			//>
+			//> x_inverse_mod_p_split = (x_inverse_mod_p & ((1 << 128) - 1), x_inverse_mod_p >> 128)
+			//>
+			//> ids.x_inverse_mod_p.low = x_inverse_mod_p_split[0]
+			//> ids.x_inverse_mod_p.high = x_inverse_mod_p_split[1]
+			pack512 := func(lolow, loHigh, hiLow, hiHigh *fp.Element, numBitsShift int) big.Int {
+				var loLowBig, loHighBig, hiLowBig, hiHighBig big.Int
+				lolow.BigInt(&loLowBig)
+				loHigh.BigInt(&loHighBig)
+				hiLow.BigInt(&hiLowBig)
+				hiHigh.BigInt(&hiHighBig)
+
+				return *new(big.Int).Add(new(big.Int).Lsh(&hiHighBig, uint(numBitsShift)), &loLowBig).Add(new(big.Int).Lsh(&hiLowBig, uint(numBitsShift)), &loHighBig)
+			}
+			pack := func(low, high *fp.Element, numBitsShift int) big.Int {
```

Comment:
do we need a separate function for that? I would just put this code right above the packing of the two variables, and not make an anonymous function out of that